base/gocb_utils.go

```diff
@@ -153,3 +158,52 @@ func getRootCAs(caCertPath string) (*x509.CertPool, error) {
 	}
 	return rootCAs, nil
 }
+
+// waitForGoCB response waits for a response from a gocb.Agent
+func waitForGoCBResponse(signal <-chan error, op gocbcore.PendingOp, timeout time.Duration) error {
```

Comment:
We've got some existing code (like Collection.GetStatsVbSeqno) that does the gocb callback handling in what I think is a bit simpler/cleaner implementation.  Have a look and let me know if you think that approach can be followed here.

Reply:
This is unfortunately not possible due to GOCBC-1332.